### PR TITLE
Oct 2299 BIS headset to sync to multiple BISes and output the audio for one of these

### PR DESCRIPTION
**Pull request by gWacey**
As a user of the BLE broadcast audio I would like to synchronise to a number of BISes and stream one of these to the output within the earbud. At some point, on a button press, I would like to switch to a different BIS synchronised to and output the audio for this new stream.

### CODE REVIEW

File: applications/nrf5340_audio/tools/buildprog/nrf5340_audio_dk_devices.json [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-        "nrf5340_audio_dk_snr": 1000,
-        "nrf5340_audio_dk_dev": "gateway",
-        "channel": "NA"
- },
- {
-        "nrf5340_audio_dk_snr": 1000,
-        "nrf5340_audio_dk_dev": "headset",
-        "channel": "right"
- }
+       {
```
Comment (koffes) [inline applications/nrf5340_audio/tools/buildprog/nrf5340_audio_dk_devices.json:2]: All this needs to be removed. You can ignore it locally.

**Review by koffes — CHANGES_REQUESTED** [review]

**Review by koffes — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/audio/streamctrl.c [hunk 2]
```diff
@@ -300,38 +333,20 @@ static void button_evt_handler(struct button_evt event)
 
 		break;
 
-	case BUTTON_TEST_TONE:
+	case BUTTON_4:
 		if (IS_ENABLED(CONFIG_WALKIE_TALKIE_DEMO)) {
-			LOG_DBG("Test tone not supported in walkie-talkie mode");
-			return;
+			LOG_DBG("Button functions not supported in walkie-talkie mode");
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/audio/streamctrl.c:338]: Hmm, for BUTTON_PLAY_PAUSE DBG is used

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 3]
```diff
@@ -46,27 +77,39 @@ static bool playing_state = true;
...
+		LOG_INF("Retrive the codec configuration for LC3:");
+		codec_info->id = codec->id;
+		codec_info->cid = codec->cid;
+		codec_info->vid = codec->vid;
+		codec_info->frequency = bt_codec_cfg_get_freq(codec);
+		codec_info->frame_duration_us = bt_codec_cfg_get_frame_duration_us(codec);
+		bt_codec_cfg_get_chan_allocation_val(codec, &codec_info->chan_allocation);
+		codec_info->octets_per_sdu = bt_codec_cfg_get_octets_per_frame(codec);
+		codec_info->bitrate =
+			codec_info->octets_per_sdu * 8 * (1000000 / codec_info->frame_duration_us);
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:109]: Please check my change here

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 4]
```diff
@@ -46,27 +77,39 @@ static bool playing_state = true;
...
-			LOG_INF("\tChannel allocation: 0x%x", chan_allocation);
+		LOG_INF("\tFrequency: %d Hz", codec->frequency);
+		LOG_INF("\tFrame Duration: %d us", codec->frame_duration_us);
+		if (codec->chan_allocation >= 0) {
+			LOG_INF("\tChannel allocation: 0x%x", codec->chan_allocation);
 		}
+		LOG_INF("\tOctets per frame: %u (%u kbps)", codec->octets_per_sdu, codec->bitrate);
+		LOG_INF("\tFrames per SDU: %d", codec->blocks_per_sdu);
+	} else {
+		LOG_WRN("Codec is not LC3, codec_id: 0x%2hhx", codec->id);
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:92]: @gWacey

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 5]
```diff
@@ -208,63 +251,72 @@ static void base_recv_cb(struct bt_audio_broadcast_sink *sink, const struct bt_a
...
-	LOG_DBG("Received BASE with %d subgroup(s) from broadcast sink", base->subgroup_count);
+	sync_stream_cnt = 0;
 
 	/* Search each subgroup for the BIS of interest */
 	for (int i = 0; i < base->subgroup_count; i++) {
 		for (int j = 0; j < base->subgroups[i].bis_count; j++) {
 			const uint8_t index = base->subgroups[i].bis_data[j].index;
 
-			LOG_DBG("BIS %d   index = %d", j, index);
+			LOG_DBG("BIS %d   index = %hhu", j, index);
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:268]: I have changed this section abit now, we can have a walkthrough

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 6]
```diff
@@ -208,63 +251,72 @@ static void base_recv_cb(struct bt_audio_broadcast_sink *sink, const struct bt_a
...
-						(struct bt_codec *)&base->subgroups[i].codec;
-					print_codec(streams[i].codec);
+				LOG_DBG("Stream %u in subgroup %d from broadcast sink",
+					sync_stream_cnt, i);
 
-					LOG_DBG("Stream %d in subgroup %d from broadcast sink", i,
-						j);
+				sync_stream_cnt += 1;
+				if (sync_stream_cnt >= ARRAY_SIZE(audio_streams)) {
+					break;
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:284]: Lets discuss this in the walkthrough

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 7]
```diff
@@ -208,63 +251,72 @@ static void base_recv_cb(struct bt_audio_broadcast_sink *sink, const struct bt_a
...
+
 	if (base_bis_index_bitfield) {
+		bis_index_bitfield = base_bis_index_bitfield & bis_index_mask;
+
 		ret = ctrl_events_le_audio_event_send(LE_AUDIO_EVT_CONFIG_RECEIVED);
 		ERR_CHK(ret);
-		bis_index_bitfield = base_bis_index_bitfield & bis_index_mask;
 
+		LOG_DBG("Channel %s active from bit mask %u",
+			((active_stream_index == AUDIO_CH_L) ? CH_L_TAG : CH_R_TAG),
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:305]: See my added comment in the code. It think we can keep it like this for now

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 8]
```diff
@@ -364,20 +421,56 @@ static int bis_headset_cleanup(bool from_sync_lost_cb)
 	return 0;
 }
 
+int le_audio_user_defined_button_press(void)
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:424]: We can discuss in the walkthrough

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 9]
```diff
@@ -386,7 +479,7 @@ int le_audio_volume_up(void)
 
 int le_audio_volume_down(void)
 {
-	if (streams[0].ep->status.state != BT_AUDIO_EP_STATE_STREAMING) {
+	if (active_audio_stream.stream->ep->status.state != BT_AUDIO_EP_STATE_STREAMING) {
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:482]: I agree that it is abit strange. But I cannot come up with a better name. It is a "bt_audio_stream", not e.g. ID or a number

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 10]
```diff
@@ -364,20 +421,56 @@ static int bis_headset_cleanup(bool from_sync_lost_cb)
 	return 0;
 }
 
+int le_audio_user_defined_button_press(void)
+{
+	int ret = 0;
+	unsigned int stream_index;
+
+	playing_state = false;
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:429]: Looks like her is fine

**Comment by erikrobstad** [comment]
Replaced by https://github.com/nrfconnect/sdk-nrf/pull/9428

**Pull request closed without merging**